tests/unit/authors/test_authors_dojson.py

```diff
@@ -0,0 +1,479 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of INSPIRE.
+# Copyright (C) 2016 CERN.
+#
+# INSPIRE is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# INSPIRE is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with INSPIRE. If not, see <http://www.gnu.org/licenses/>.
+#
+# In applying this licence, CERN does not waive the privileges and immunities
+# granted to it by virtue of its status as an Intergovernmental Organization
+# or submit itself to any jurisdiction.
+
+from __future__ import absolute_import, division, print_function
+
+from dojson.utils import GroupableOrderedDict
+
+from inspirehep.modules.authors.dojson.model import updateform
+
+
+def test_name_status_from_status():
+    form = GroupableOrderedDict([
+        ('status', 'foo'),
+    ])
+
+    expected = {
+        'status': 'foo',
+    }
+    result = updateform.do(form)
+
+    assert expected == result['name']
+
+
+def test_name_status_from_status_updates():
+    form = GroupableOrderedDict([
+        ('status', 'foo'),
+        ('status', 'bar'),
+    ])
+
+    expected = {
+        'status': 'bar',
+    }
+    result = updateform.do(form)
+
+    assert expected == result['name']
+
+
+def test_name_preferred_name_from_display_name():
+    form = GroupableOrderedDict([
+        ('display_name', 'foo'),
+    ])
+
+    expected = {
+        'preferred_name': 'foo',
+    }
+    result = updateform.do(form)
+
+    assert expected == result['name']
+
+
+def test_name_preferred_name_from_display_name_updates():
+    form = GroupableOrderedDict([
+        ('display_name', 'foo'),
+        ('display_name', 'bar'),
+    ])
+
+    expected = {
+        'preferred_name': 'bar',
+    }
+    result = updateform.do(form)
+
+    assert expected == result['name']
+
+
+def test_urls_from_websites_webpage():
+    form = {
+        'websites': [
+            {
+                'webpage': 'foo',
+            },
+        ],
+    }
```

Comment:
This doesn't use a `GroupableOrderedDict` because I couldn't figure out the syntax to have a value of type `list`.